core/src/main/java/fr/sncf/osrd/conflicts/SpacingResourceGenerator.kt

```diff
@@ -78,8 +76,8 @@ class SpacingRequirementAutomaton(
     val loadedSignalInfra: LoadedSignalInfra,
     val blockInfra: BlockInfra,
     val simulator: SignalingSimulator,
-    val callbacks: IncrementalRequirementCallbacks,
-    val incrementalPath: IncrementalPath,
+    var callbacks: IncrementalRequirementCallbacks, // Not read-only to be updated along the path
+    var incrementalPath: IncrementalPath, //
```

Comment:
Guessing you forgot the comment here.
```suggestion
    var incrementalPath: IncrementalPath, // Not read-only to be updated along the path
```